src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,167 @@
+  align-items: center;
+  overflow: hidden;
+  transition: all 100ms ease-in-out;
+  width: 100%;
+  width: -webkit-fill-available;
+  width: fill-available;
+  width: stretch;
+  min-height: 2px;
+  ${({ theme, $completed, $progress, $type, $error }) => `
+    background: ${
+      $completed
+        ? theme.click.field.color.background.default
+        : `linear-gradient(to right, ${theme.global.color.accent.default} 0%, ${theme.global.color.accent.default} ${$progress}%, ${theme.click.field.color.background.default} ${$progress}%,  ${theme.click.field.color.background.default} 100%)`
+    };
+    background-size: calc(100% + 2px);
+    background-position: center;
+    gap: ${theme.click.field.space.gap};
+    border-radius: ${theme.click.field.radii.all};
+    font: ${theme.typography.styles.product.text.normal.sm};
+    padding: ${$type === "default" ? "0.25rem" : 0} ${theme.click.field.space.x};
+    padding-right: 0;
+    color: ${theme.global.color.accent.default};
+    border: 1px solid ${theme.click.field.color.stroke.default};
+    &:hover {
+      border: 1px solid ${theme.click.field.color.stroke.hover};
+      background: ${
+        $completed
+          ? theme.click.field.color.background.hover
+          : `linear-gradient(to right, ${theme.global.color.accent.default} 0%, ${theme.global.color.accent.default} ${$progress}%, ${theme.click.field.color.background.hover} ${$progress}%,  ${theme.click.field.color.background.hover} 100%)`
+      };
+      background-size: calc(100% + 2px);
+      background-position: center;
+    }
+    &:focus-within, &:focus {
+      border: 1px solid ${theme.click.field.color.stroke.active};
+      background: ${
+        $completed
+          ? theme.click.field.color.background.active
+          : `linear-gradient(to right, ${theme.global.color.accent.default} 0%, ${theme.global.color.accent.default} ${$progress}%, ${theme.click.field.color.background.active} ${$progress}%,  ${theme.click.field.color.background.active} 100%)`
+      };
+      background-size: calc(100% + 2px);
+      background-position: center;
+    }
+    ${$error ? `border-color: ${theme.click.field.color.stroke.error};` : ""}
+  `};
+`;
+
+const ProgressText = styled.span<{ $completed: boolean }>`
+  width: 100%;
+  font: inherit;
+  mix-blend-mode: difference;
+`;
+
+const ProgressCloseButton = styled.button<{ $dismissable?: boolean }>`
+  mix-blend-mode: difference;
+  border: 0;
+  visibility: ${({ $dismissable }) => ($dismissable ? "visible" : "hidden")};
+`;
+
+const FormRootWrapper = styled(FormRoot)`
+  align-items: ${({ $orientation }) =>
+    $orientation === "horizontal" ? "center" : "flex-start"};
+`;
+
+export const ProgressBar = ({
+  progress,
+  type = "default",
+  dismissable = false,
+  onCancel,
+  successMessage,
+  label,
+  id,
+  orientation,
+  error,
+  dir,
+  ...props
+}: ProgressBarProps) => {
+  const completed = progress === 100;
+  const defaultId = useId();
+  return (
+    <FormRootWrapper
```

Comment:
why is this inside a form component? Isn't this just a graphic component?

Reply:
The new design has a label component inside the dataloading figma

Reply:
I still think this should live on its own. But we should support to compose it with label or not.
Would really not tie this with anything form related.

Reply:
I think the mail purpose was the padding and styling of the elements